File: SistemaTicket/Entities/Ticket.cs
using SistemaTicket.Enums;
using System.ComponentModel.DataAnnotations;

namespace SistemaTicket.Entities;

public class Ticket
{
    public int Id { get; set; }
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public TicketStatus Status { get; set; }
    public TicketPriority Priority { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public string CreatedById { get; set; } = string.Empty;
    public ApplicationUser CreatedBy { get; set; } = null!;
    public string? AssignedToId { get; set; }
    public ApplicationUser? AssignedTo { get; set; }
    public List<TicketComment> TicketComments { get; set; } = [];
    public List<TicketHistory> TicketHistories { get; set; } = [];
}

File: frontend/src/types/ticket.ts
import type { TicketComment } from './TicketComment';
import type { TicketHistory } from './ticketHistory';

export const TICKET_STATUS = {
  OPEN: 'Open',
  INPROGRESS: 'InProgress',
  CLOSED: 'Closed',
} as const;

export type TicketStatus = (typeof TICKET_STATUS)[keyof typeof TICKET_STATUS];

export const TICKET_PRIORITY = {
  LOW: 'Low',
  MEDIUM: 'Medium',
  HIGH: 'High',
} as const;

export type TicketPriority =
  (typeof TICKET_PRIORITY)[keyof typeof TICKET_PRIORITY];

export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: TicketStatus;
  priority: TicketPriority;
  createdAt: string;
  createdById: string;
  createdByName: string;
  assignedToId?: string;
  assignedToName?: string;
}

export interface PagedTickets {
  tickets: Ticket[];
  total: number;
  statusCounts: StatusCounts;
}

export type StatusCounts = Record<Uncapitalize<TicketStatus>, number>;

export interface TicketDetails {
  ticket: Ticket;
  ticketComments: TicketComment[];
  ticketHistories: TicketHistory[];
}
